src/script/miniscript.h

```diff
@@ -1378,7 +1378,7 @@ inline NodeRef<Key> Parse(Span<const char> in, const Ctx& ctx)
     assert(constructed.size() == 1);
     assert(constructed[0]->ScriptSize() == script_size);
     if (in.size() > 0) return {};
-    const NodeRef<Key> tl_node = std::move(constructed.front());
```

Comment:
cc @sipa @darosior

Reply:
TIL. My understanding is that without this patch `tl_node` would not be moved a couple lines below when returning from the function. Making it not `const` fixes this.

If my understanding is correct, ACK.

Reply:
To be precise, this patch makes a compiler to call a move-constructor instead of a copy-constructor in the `return` statement.

Reply:
`NodeRef` is `std::shared_ptr<bla>`, so this shouldn't change the compiled binary except for a removed `std::atomic` add and sub.